src/chunk.c

```diff
@@ -1720,36 +1805,50 @@ ts_chunk_exists_relid(Oid relid)
 	return ts_chunk_get_by_relid(relid, 0, false) != NULL;
 }
 
-static ScanTupleResult
-chunk_tuple_delete(TupleInfo *ti, void *data)
+/* Delete the chunk tuple.
+ *
+ * mark_as_dropped_only - instead of deleting the row, mark it as dropped.
+ * this is used when we need to preserve catalog information about the chunk
+ * after dropping it. Currently only used when preserving continuous aggregates
+ * on the chunk after the raw data was dropped. Otherwise, we'd have dangling
+ * chunk ids left over in the materialization table. Preserve the space dimension
+ * info about these chunks too.
+ */
+static void
+chunk_tuple_delete(TupleInfo *ti, DropBehavior behavior, bool preserve_chunk_catalog_row)
 {
 	FormData_chunk form;
 	CatalogSecurityContext sec_ctx;
 	ChunkConstraints *ccs = ts_chunk_constraints_alloc(2, ti->mctx);
 	int i;
-	DropBehavior *behavior = data;
 
 	chunk_formdata_fill(&form, ti->tuple, ti->desc);
-	ts_chunk_constraint_delete_by_chunk_id(form.id, ccs);
-	ts_chunk_index_delete_by_chunk_id(form.id, true);
-	ts_compression_chunk_size_delete(form.id);
 
-	/* Check for dimension slices that are orphaned by the chunk deletion */
-	for (i = 0; i < ccs->num_constraints; i++)
+	/* if only marking as deleted, keep the constraints and dimension info */
+	if (!preserve_chunk_catalog_row)
 	{
-		ChunkConstraint *cc = &ccs->constraints[i];
+		ts_chunk_constraint_delete_by_chunk_id(form.id, ccs);
```

Comment:
are we skipping over these chunks during query planning? Does it happen automatically because we use the postgres tables and not our catalogs for expansion?

Reply:
yes. This is skipped in planning and tested